### PR TITLE
Adhere to pydocstyle and remove pylint errors.

### PR DESCRIPTION
**Pull request by devyanikota**
Signed-off-by: Devyani Kota <devyanikota@gmail.com>

### CODE REVIEW

File: test/api_test.py [hunk 1]
```diff
@@ -10,39 +11,30 @@
 
 
 class GlusterdConfig(object):
-    """
-    GlusterdConfig Class
-    """
+    """GlusterdConfig Class."""
```
Comment (sac) [inline test/api_test.py]: @devyanikota this doc string can be better written. Something like "Class to test the methods." or "Class to test the bindings" ...

File: test/api_test.py [hunk 2]
```diff
@@ -80,18 +68,14 @@ def test_peer_add(gd2client):
...
     status, _ = gd2client.peer_status()
     # assert resp[0]['id'] == Peer.get_peer_id)
     assert status == 200
 
 
 def test_peer_remove(gd2client):
-    """
-    This function tests for peer remove.
-    """
+    """Tests for peer remove."""
```
Comment (sac) [inline test/api_test.py]: Nit: Not sure if this is grammatically correct. Should it be: Test peer removal. Or Test for peer removal. 

I'm actually not sure, I may be wrong.

Comment (devyanikota) [reply test/api_test.py]: I think I had implied, "Tests for peer remove operation" but "Test peer removal" is more precise.
@sac Thanks.

File: glusterapi/peer.py [hunk 3]
```diff
@@ -27,19 +27,19 @@ def peer_add(self, host, metadata=None, zone=""):
 
     def peer_remove(self, peerid):
         """
-        Gluster Peer Remove
+        Gluster Peer Remove.
 
         :param host: (string) Hostname or IP
```
Comment (Madhu-1) [inline glusterapi/peer.py]: can you change this function comment, we cannot remove peer from Hostname or IP, a peer can be removed only by using peer id.

Comment (devyanikota) [reply glusterapi/peer.py]: done.

File: glusterapi/peer.py [hunk 4]
```diff
@@ -8,10 +8,10 @@
 class PeerApis(BaseAPI):
     def peer_add(self, host, metadata=None, zone=""):
         """
-        Gluster Peer Add
+        Gluster Peer Add.
```
Comment (Madhu-1) [inline glusterapi/peer.py]: sorry missed checking this one.
you need to add function document for metadata and zone also

File: test/api_test.py [hunk 5]
```diff
@@ -80,18 +68,14 @@ def test_peer_add(gd2client):
 
 
 def test_peer_status(gd2client):
-    """
-    This function tests for peer status.
-    """
+    """Test for peer status."""
     status, _ = gd2client.peer_status()
     # assert resp[0]['id'] == Peer.get_peer_id)
```
Comment (Madhu-1) [inline test/api_test.py]: remove commented line

File: glusterapi/peer.py [hunk 6]
```diff
@@ -1,17 +1,19 @@
...
 
 class PeerApis(BaseAPI):
     def peer_add(self, host, metadata=None, zone=""):
         """
-        Gluster Peer Add
+        Gluster Peer Add.
 
         :param host: (string) Hostname or IP
-        :raises: GlusterApiError on failure
+        :param metadata: (dictionary of string: string) peer info mapping
```
Comment (Madhu-1) [inline glusterapi/peer.py]: you can change this to something like

```custom key/value pairs (metadata) for peers```

Comment (devyanikota) [reply glusterapi/peer.py]: done.

File: test/api_test.py (hunk 5 above)
Comment (devyanikota) [reply test/api_test.py]: done

**Pull request merged**